### PR TITLE
Fix: Rename docs to catalog to keep routes consistent

### PR DESCRIPTION
**Pull request by mykalmax**
- Changing related routes from `docs` to `data-catalog`
- both cli commands `sqlmesh ui --mode docs` and `sqlmesh ui --mode catalog` spin up SQLMesh UI in `docs`/ `catalog` mode but with routing name `data-catalog``
- Changing internal references form `Docs` to `Data Catalog`

### CODE REVIEW

File: web/client/src/routes.tsx [hunk 1]
```diff
@@ -419,5 +419,7 @@ export function getBrowserRouter(
     },
   ].filter(Boolean) as RouteObject[]
 
+  console.log('routes', routes, getUrlPrefix())
+
```
Comment (vchan) [inline web/client/src/routes.tsx:423]: Debugging?

**Review by vchan — APPROVED** [review]
Does our documentation need to be updated to reflect the renaming of docs to catalog?

**Comment by mykalmax** [comment]
> Does our documentation need to be updated to reflect the renaming of docs to catalog?

so `sqlmesh ui --mode docs` should work as before and change mostly internal, and i don't think we have any references to`/docs`  url

**Comment by RN-nguyen** [comment]
Verified opening sqlmesh ui/docs/catalog routes to correct page.

File: docs/guides/ui.md [hunk 2]
```diff
@@ -232,9 +232,10 @@ You may specify the UI mode as an option when you [start the UI on the command l
 
 The UI modes contain these modules:
 
-- `editor`: code editor, plan builder, project documentation, table and column lineage
-- `plan`: plan builder, project documentation, table and column lineage
-- `docs`: project documentation, table and column lineage
+- `editor`: code editor, plan builder, data catalog, table and column lineage
+- `plan`: plan builder, data catalog, table and column lineage
+- `docs`: data catalog, table and column lineage
```
Comment (treysp) [inline docs/guides/ui.md:237]: Should this be replaced by the `catalog` entry on the next line?

File: sqlmesh/cli/main.py [hunk 3]
```diff
@@ -696,9 +696,9 @@ def info(obj: Context, skip_connection: bool) -> None:
 )
 @click.option(
     "--mode",
-    type=click.Choice(["ide", "default", "docs", "plan"], case_sensitive=False),
-    default="default",
-    help="Mode to start the UI in. Default: default",
+    type=click.Choice(["ide", "catalog", "docs", "plan"], case_sensitive=False),
```
Comment (treysp) [inline sqlmesh/cli/main.py:699]: So `docs` is now an alias for `catalog`?

**Review by treysp — APPROVED** [review]

File: sqlmesh/cli/main.py (hunk 3 above)
Comment (mykalmax) [reply sqlmesh/cli/main.py:699]: yes, so that old `--mode docs` keeps working

**Pull request merged**